File: task1/task1b/Customer.cpp
#include "Customer.h"
#include <iostream>


Customer::Customer(){}

Customer::Customer(int id, string name, string road, string town, string postCode): customerID(id), customerName(name), customerRoad(road), customerTown(town), customerPostCode(postCode){}

void Customer::display_customer_info(){
    cout << "Customer ID: " << getID() << endl;
    cout << "Customer Name: " << getName() << endl;
    cout << "Customer Address: " << getRoad() << ", " << getTown() << ", " << getPostCode() << endl;
}

// Setters
void Customer::setID(int id){
    customerID = id;
}

void Customer::setName(const string& name){
    customerName = name;
}

void Customer::setRoad(const string& road){
    customerRoad = road;
}

void Customer::setTown(const string& town){
    customerTown = town;
}

void Customer::setPostCode(const string& postCode){
    customerPostCode = postCode;
}

// Getters
int Customer::getID() const{
    return customerID;
}

string Customer::getName() const{
    return customerName;
}

string Customer::getRoad() const{
    return customerRoad;
}

string Customer::getTown() const{
    return customerTown;
}

string Customer::getPostCode() const{
    return customerPostCode;
}

File: task1/task1c/src/main/java/com/task1c/Customer.java
package com.task1c;

import java.io.Serializable;

public class Customer implements Serializable{
    private static final long serialVersionUID = 1L;

    protected int customerID;
    protected String customerName;
    protected String customerRoad, customerTown, customerPostCode;

    public Customer(){}

    public Customer(int id, String name, String road, String town, String postCode){
        customerID = id;
        customerName = name;
        customerRoad = road;
        customerTown = town;
        customerPostCode = postCode;
    }

    public String toString(){
        String info = "";

        info += "Customer ID: " + getID() + "\n";
        info += "Customer Name: " + getName() + "\n";
        info += "Customer Address: " + getRoad() + ", " + getTown() + ", " + getPostCode() + "\n";

        return info;
    }

    // Setters
    public void setID(int id){
        customerID = id;
    }

    public void setName(String name){
        customerName = name;
    }

    public void setRoad(String road){
        customerRoad = road;
    }

    public void setTown(String town){
        customerTown = town;
    }

    public void setPostCode(String postCode){
        customerPostCode = postCode;
    }

    // Getters
    public int getID(){
        return customerID;
    }

    public String getName(){
        return customerName;
    }

    public String getRoad(){
        return customerRoad;
    }

    public String getTown(){
        return customerTown;
    }

    public String getPostCode(){
        return customerPostCode;
    }
}
